iModelCore/iModelPlatform/Tests/DgnProject/Compatibility/IModelCompatibilityTests.cpp

```diff
@@ -2071,3 +2071,200 @@ TEST_F(IModelCompatibilityTestFixture, UpgradeDomainIModelToEC32)
 
     IModelEvolutionTestsDomain::GetDomain().SetRequired(DgnDomain::Required::No);
     }
+
+TEST_F(IModelCompatibilityTestFixture, MajorSchemaUpgradeDeletePropertyAndClass)
+    {
+    for (const auto& testFile : DgnDbProfile::Get().GetAllVersionsOfTestFile(TESTIMODEL_EMPTY))
+        {
+        for (std::unique_ptr<TestIModel> testDbPtr : TestIModel::GetPermutationsFor(testFile))
+            {
+            TestIModel& testDb = *testDbPtr;
+            const auto openStat = testDb.Open();
+            const auto& params = static_cast<DgnDb::OpenParams const&> (testDb.GetOpenParams());
+
+            if (params.IsReadonly())
+                continue;
+
+            //schema import is possible to newer ECDb profile files or files that don't support EC3.2 as the profile version is automatically upgraded
+            ASSERT_EQ(BE_SQLITE_OK, openStat) << testDb.GetDescription();
+
+            DgnDbR dgnDb = testDb.GetDgnDb();
+
+            // Setup test schema
+            auto deserializationCtx = TestFileCreator::DeserializeSchema(testDbPtr->GetDb(), SchemaItem(R"xml(<?xml version="1.0" encoding="utf-8" ?>
+                            <ECSchema schemaName="TestSchema" alias="ts" version="1.0" xmlns="http://www.bentley.com/schemas/Bentley.ECXML.3.1">

```

Comment:
We need a test where the property has data, and the class has instances. Also, re-add the same property and class again after deleting. Verify if data for the property was set to NULL and instances of the class do not exist or class was not deleted if there were instances of that class.

Reply:
Done !